assets/src/edit-story/app/media/mediaProvider.js

```diff
@@ -91,6 +104,110 @@ function MediaProvider({ children }) {
     ]
   );
 
+  const resetMedia = useCallback(
```

Comment:
Wouldn't `resetWithFetch` does this?

Reply:
Hey @spacedmonkey good point, anyway, I even tried `resetWithFetch` and `fetchMedia` directly before proposing this change, but after catching up on several issues related to duplication on the Media Library listing, it was really necessary. 

In my investigations, I discover that this code https://github.com/google/web-stories-wp/blob/master/assets/src/edit-story/app/media/useMediaReducer/reducer.js#L51  was the problem (e.g. mainly the declaration `[...state.media, ...media]`). 

As far as I understand, this code was changed in order to support infinite scroll somewhat. So when I was re-fetching data using `fetchMedia` or `resetWithFetch`, both were duplicating local/remote media in race conditions since that code creates a new List reusing the old list state. So my solution to set a fresh `media` state is a proposal to always have a solid and updated data in the media store. Let me know if it clarifies, also, if you have some additional improvement or even a short way in mind, I would love to hear. Thanks for point this out.

Reply:
A comment explaining this is the code might be useful for future devs.